store/src/main/java/com/book/store/service/impl/BookServiceImpl.java

```diff
@@ -12,13 +16,32 @@
 public class BookServiceImpl implements BookService {
     private BookRepository bookRepository;
 
+    private BookMapper bookMapper;
+
     @Override
     public Book save(Book book) {
         return bookRepository.save(book);
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public List<BookDto> getAllBooks() {
+        List<Book> books = bookRepository.findAll();
+        return bookMapper.toBookDtoList(books);
+    }
+
+    @Override
+    public BookDto getBookById(Long id) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() ->
+                        new EntityNotFoundException("Book with id " + id + " not found."));
+        return bookMapper.toBookDto(book);
```

Comment:
Use Optional#map to reduce code.
```suggestion
        Book book = bookRepository.findById(id)
                .orElseThrow(() ->
                        new EntityNotFoundException("Book with id " + id + " not found."));
        return bookMapper.toBookDto(book);
```